### PR TITLE
3083 asset catalogue filtering

### LINKED ISSUE #3083: Better filtering for catalogue list view
## Is your feature request related to a problem? Please describe 👀
An extension for #2956  - implement lookups for class, category and type so that the filters can use the ids and function correctly.

The hierarchy of the filters needs to be considered too.

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #3083

# 👩🏻‍💻 What does this PR do? 
 
Adds the queries for categories, types and classes and uses those to populate the filters.
You can filter by all the available options now, with the types dependent on the category selected. 
Haven't implemented the classes yet, as we only have one and that's a bit silly.

Screenshot 2024-02-27 at 5 49 47 PM
Screenshot 2024-02-27 at 5 50 21 PM
Screenshot 2024-02-27 at 5 50 29 PM

# 🧪 How has/should this change been tested? 

## 📃 Documentation

- [ ] Describe the filtering

### CODE REVIEW

File: client/packages/common/src/index.ts [hunk 1]
```diff
@@ -17,7 +17,6 @@ export {
   useQuery,
   UseQueryResult,
   QueryClient,
-  QueryClientProvider,
```
Comment (mark-prins) [inline client/packages/common/src/index.ts:20]: This breaks plugins! not sure why it was added back in here.

File: client/packages/common/src/ui/components/inputs/Filters/FilterMenu.tsx [hunk 2]
```diff
@@ -90,6 +85,21 @@ export const FilterMenu: FC<FilterDefinitions> = ({ filters }) => {
 
   const showRemoveOption = activeFilters.length > 0;
 
+  // updating active filters when the filters are changed
```
Comment (mark-prins) [inline client/packages/common/src/ui/components/inputs/Filters/FilterMenu.tsx:88]: as noted. required to have the filters update when they are changed, such as when new options are provided

File: client/packages/system/src/Asset/ListView/ListView.tsx [hunk 3]
```diff
@@ -40,7 +40,7 @@ const AssetListComponent: FC = () => {
       {
         key: 'manufacturer',
         Cell: TooltipTextCell,
-        maxWidth: 350,
+        width: 350,
```
Comment (mark-prins) [inline client/packages/system/src/Asset/ListView/ListView.tsx:43]: to stop the columns moving around when you scroll (due to the ViewPort and changing data values in the rows)

File: client/packages/common/src/types/schema.ts [hunk 4]
```diff
@@ -336,7 +336,7 @@ export type AssetTypeFilterInput = {
 
 export type AssetTypeNode = {
   __typename: 'AssetTypeNode';
-  catetoryId: Scalars['String']['output'];
```
Comment (mark-prins) [inline client/packages/common/src/types/schema.ts:339]: offroad: typo

File: client/packages/system/src/Asset/api/operations.graphql [hunk 5]
```diff
@@ -51,3 +51,41 @@ query assetCatalogueItemById(
     }
   }
 }
+
+query assetClasses($storeId: String!, $sort: AssetClassSortInput) {
+  assetClasses(storeId: $storeId, sort: $sort) {
+    ... on AssetClassConnector {
+      nodes {
+        name
+        id
```
Comment (jmbrunskill) [inline client/packages/system/src/Asset/api/operations.graphql:60]:
```suggestion
        id
        name
```
not: consistent ordering

File: client/packages/system/src/Asset/api/operations.graphql [hunk 6]
```diff
@@ -51,3 +51,41 @@ query assetCatalogueItemById(
...
+}
+
+query assetCategories($storeId: String!, $sort: AssetCategorySortInput) {
+  assetCategories(storeId: $storeId, sort: $sort) {
+    ... on AssetCategoryConnector {
+      nodes {
+        classId
+        id
+        name
+      }
```
Comment (jmbrunskill) [inline client/packages/system/src/Asset/api/operations.graphql:87]:
```suggestion
      nodes {
        id
        name
        classId
      }
```
nit: consistency with ordering...

File: client/packages/system/src/Patient/Encounter/CreateEncounterModal.tsx
Comment (jmbrunskill) [inline client/packages/system/src/Patient/Encounter/CreateEncounterModal.tsx:1]: Move stuff from develop?

**Review by jmbrunskill — APPROVED** [review]
Looks good to me, looks like we have some stuff from develop? Assuming your confident that's correct, I'm happy :)

File: client/packages/common/src/index.ts (hunk 1 above)
Comment (mark-prins) [reply client/packages/common/src/index.ts:20]: not yet, I think. Just needed to get some screenshots for a proposal. I'm wondering about creating the dashboard widgets solely as plugins and having them configurable and if we get to that point, the plugin framework is tested by everyone!

File: client/packages/inventory/src/Stocktake/api/hooks/line/useStocktakeLines.ts [hunk 7]
```diff
@@ -1,16 +1,16 @@
 import { useQuery, useUrlQueryParams } from '@openmsupply-client/common';
 import { useStocktakeApi } from '../utils/useStocktakeApi';
+import { useStocktakeNumber } from '../document/useStocktake';
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply client/packages/inventory/src/Stocktake/api/hooks/line/useStocktakeLines.ts:3]: gah! it is. more merge troubles. I'll revert these changes before merging.
Thanks for picking it up

**Comment by mark-prins** [comment]
Thanks @jmbrunskill 
I've taken the commits from this branch and applied directly, to remove the develop commits. will close off this PR manually

**Pull request closed without merging**